palisade-service/src/unit-tests/java/uk/gov/gchq/palisade/service/palisade/service/PalisadeServiceTest.java

```diff
@@ -56,19 +55,37 @@ void setUp() {
     @Test
     void testRegisterDataRequest() {
         CompletableFuture<String> token = service.registerDataRequest(PALISADE_REQUEST);
-        assertThat(token.join()).isEqualTo(COMMON_UUID.toString());
-        assertThat(sinkCollection).usingRecursiveComparison().isEqualTo(List.of(new TokenRequestPair(token.join(), AUDITABLE_PALISADE_REQUEST)));
+
+        assertAll(
+                () -> assertThat(token.join())
+                        .as("Check the value of the returned Token")
+                        .isEqualTo(COMMON_UUID.toString()),
+                () -> assertThat(sinkCollection)
+                        .as("Check that the sinkCollection has only 1 TokenRequestPair, containing the token and an AuditablePalisadeSystemResponse")
+                        .usingRecursiveComparison()
+                        .asList()
+                        .containsOnly(new TokenRequestPair(token.join(), AUDITABLE_PALISADE_REQUEST))
+        );
     }
 
     @Test
     void testErrorMessage() {
         CompletableFuture<String> token = service.registerDataRequest(PALISADE_REQUEST);
         service.errorMessage(PALISADE_REQUEST, token.join(), attributes, ERROR);
-        assertThat(token.join()).isEqualTo(COMMON_UUID.toString());
         assertAll(
-                () -> assertThat(sinkCollection.getLast().first()).isEqualTo(token.join()),
-                () -> assertThat(sinkCollection.getLast().second().getAuditErrorMessage()).usingRecursiveComparison().ignoringFields("timestamp")
-                        .isEqualTo(List.of(new TokenRequestPair(token.join(), AUDITABLE_PALISADE_ERROR)).get(0).second().getAuditErrorMessage())
+                () -> assertThat(token.join())
+                        .as("Check the value of the returned Token")
+                        .isEqualTo(COMMON_UUID.toString()),
+                () -> assertThat(sinkCollection.getLast().first())
+                        .as("Check the Token value from the error-topic")
+                        .isEqualTo(token.join()),
+                () -> assertThat(sinkCollection.getLast().second().getAuditErrorMessage())
+                        .as("Recursively check the value of the AuditErrorMessage")
+                        .usingRecursiveComparison().ignoringFields("timestamp")
+                        .isEqualTo(AUDITABLE_PALISADE_ERROR.getAuditErrorMessage()),
+                () -> assertThat(sinkCollection.getLast().second().getAuditErrorMessage().getError())
+                        .as("Check the class of the thrown exception")
+                        .isExactlyInstanceOf(Throwable.class)
```

Comment:
add a message assertion similar to:
```
                        .hasMessageContaining("An error was thrown in the Palisade Service")
```
and update the commonTestData error message to be a bit more descriptive

Reply:
Added